### PR TITLE
Publish 0.5

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/unicode-org/icu4x/issues/1546

This is ready to merge.

### CODE REVIEW

**Review by zbraniecki — APPROVED** [review]
lgtm!

**Review by dminor — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +39,7 @@
       - Implement data transformer for `Script_Extensions` map data (#1353)
       - Add `General_Category` predicate functions (#1310)
       - Implement `Grapheme_Cluster_Break`, `Word_Break`, and `Sentence_Break` Unicode properties (#1233)
+      - Add fn to return UnicodeSet for Script_Extensions (#1529)
```
Comment (sffc) [inline CHANGELOG.md:42]:
```suggestion
      - Implement `Script_Extensions` property (#1353)
      - Add `General_Category` predicate functions (#1310)
      - Implement `Grapheme_Cluster_Break`, `Word_Break`, and `Sentence_Break` Unicode properties (#1233)
```

**Review by sffc — APPROVED** [review]
Approved with a changelog suggestion which could go in another PR

**Comment by Manishearth** [comment]
Everything except for `icu_provider_cldr` and `icu_datagen` have been published. The latter two depend on list formatter being publishable

**Pull request merged**